NOnion/Utility/FSharpUtil.fs

```diff
@@ -4,94 +4,41 @@ open System
 open System.Runtime.ExceptionServices
 
 open FSharpx.Collections
+open Fsdk
 
 open NOnion
 
 module FSharpUtil =
-    //Implementation copied from https://github.com/nblockchain/geewallet/blob/master/src/GWallet.Backend/FSharpUtil.fs
-    let ReRaise(ex: Exception) : Exception =
-        (ExceptionDispatchInfo.Capture ex).Throw()
-        failwith "Should be unreachable"
-        ex
-
-    let rec public FindException<'T when 'T :> Exception>
-        (ex: Exception)
-        : Option<'T> =
-        let rec findExInSeq(sq: seq<Exception>) =
-            match Seq.tryHeadTail sq with
-            | Some(head, tail) ->
-                match FindException head with
-                | Some ex -> Some ex
-                | None -> findExInSeq <| tail
-            | None -> None
-
-        if isNull ex then
-            None
-        else
-            match ex with
-            | :? 'T as specificEx -> Some specificEx
-            | :? AggregateException as aggEx ->
-                findExInSeq aggEx.InnerExceptions
-            | _ -> FindException<'T> ex.InnerException
-
-    type private Either<'Val, 'Err when 'Err :> Exception> =
-        | FailureResult of 'Err
-        | SuccessfulValue of 'Val
-
     let WithTimeout (timeSpan: TimeSpan) (job: Async<'R>) : Async<'R> =
         async {
-            let read =
-                async {
-                    let! value = job
-                    return value |> SuccessfulValue |> Some
-                }
-
-            let delay =
-                async {
-                    let total = int timeSpan.TotalMilliseconds
-                    do! Async.Sleep total
-                    return FailureResult <| TimeoutException() |> Some
-                }
-
-            let! dummyOption = Async.Choice([ read; delay ])
+            let! result = FSharpUtil.WithTimeout timeSpan job
 
-            match dummyOption with
-            | Some theResult ->
-                match theResult with
-                | SuccessfulValue r -> return r
-                | FailureResult _ -> return raise <| TimeoutErrorException()
-            | None ->
-                // none of the jobs passed to Async.Choice returns None
-                return failwith "unreachable"
+            match result with
```

Comment:
@webwarrior-ws why keep WithTimeout here instead of directly use the one in Fsdk?

Reply:
Because of different semantics. Fsdk variant returns `option`, but NOnion's variant returns value or raises `TimeoutErrorException`.

Reply:
well, the idea of the initial request about combining both NOnion and geewallet's FSharpUtil modules into just one means that we should keep the best approach in each case; so let's decide what's the best approach and have only one (if it requires changing all callers of the func in geewallet or NOnion, that's fine)

Reply:
Changed NOnion to use Fsdk variant.